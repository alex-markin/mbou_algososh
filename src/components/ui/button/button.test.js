import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

describe('Button component', () => {
  it('renders button with text', () => {
    const { container } = render(<Button text="Click" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders button without text', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const { container } = render(<Button disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders loader when isLoader prop is true', () => {
    const { container } = render(<Button isLoader />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onClick callback when button is clicked', () => {
    const onClickMock = jest.fn();
    const { container } = render(<Button text="Click" onClick={onClickMock} />);
    expect(container.firstChild).toMatchSnapshot();

    const button = screen.getByRole('button', { name: /click/i });
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
