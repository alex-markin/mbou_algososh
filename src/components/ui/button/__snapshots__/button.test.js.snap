// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component calls onClick callback when button is clicked 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click
  </p>
</button>
`;

exports[`Button component renders button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click
  </p>
</button>
`;

exports[`Button component renders button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component renders disabled button 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component renders loader when isLoader prop is true 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
